SoObjects/SOGo/SOGoUserManager.m

```diff
@@ -698,44 +697,49 @@ - (void) _fillContactInfosForUser: (NSMutableDictionary *) currentUser
+            c_uid = [userEntry objectForKey: @"c_uid"];
           if (!c_domain)
             c_domain = [userEntry objectForKey: @"c_domain"];
-	  c_emails = [userEntry objectForKey: @"c_emails"];
-	  if ([c_emails count])
-	    [emails addObjectsFromArray: c_emails];
-	  if (!c_imaphostname)
-	    c_imaphostname = [userEntry objectForKey: @"c_imaphostname"];
+          c_emails = [userEntry objectForKey: @"c_emails"];
+          if ([c_emails count])
+            [emails addObjectsFromArray: c_emails];
+          if (!c_imaphostname)
+            c_imaphostname = [userEntry objectForKey: @"c_imaphostname"];
           if (!c_imaplogin)
             c_imaplogin = [userEntry objectForKey: @"c_imaplogin"];
           if (!c_sievehostname)
             c_sievehostname = [userEntry objectForKey: @"c_sievehostname"];
-	  access = [[userEntry objectForKey: @"CalendarAccess"] boolValue];
-	  if (!access)
-	    [currentUser setObject: [NSNumber numberWithBool: NO]
-			 forKey: @"CalendarAccess"];
-	  access = [[userEntry objectForKey: @"MailAccess"] boolValue];
-	  if (!access)
-	    [currentUser setObject: [NSNumber numberWithBool: NO]
-			 forKey: @"MailAccess"];
-	  
-	  // We check if it's a group
+          access = [[userEntry objectForKey: @"CalendarAccess"] boolValue];
+          if (!access)
+            [currentUser setObject: [NSNumber numberWithBool: NO]
+                            forKey: @"CalendarAccess"];
+          access = [[userEntry objectForKey: @"MailAccess"] boolValue];
+          if (!access)
+            [currentUser setObject: [NSNumber numberWithBool: NO]
+                            forKey: @"MailAccess"];
+
+          // We check if it's a group
           isGroup = [userEntry objectForKey: @"isGroup"];
           if (isGroup)
             [currentUser setObject: isGroup forKey: @"isGroup"];
 
-	  // We also fill the resource attributes, if any
-	  if ([userEntry objectForKey: @"isResource"])
-	    [currentUser setObject: [userEntry objectForKey: @"isResource"]
-			 forKey: @"isResource"];
-	  if ([userEntry objectForKey: @"numberOfSimultaneousBookings"])
-	    [currentUser setObject: [userEntry objectForKey: @"numberOfSimultaneousBookings"]
-			 forKey: @"numberOfSimultaneousBookings"];
+          // We also fill the resource attributes, if any
+          if ([userEntry objectForKey: @"isResource"])
+            [currentUser setObject: [userEntry objectForKey: @"isResource"]
+                            forKey: @"isResource"];
+          if ([userEntry objectForKey: @"numberOfSimultaneousBookings"])
+            [currentUser setObject: [userEntry objectForKey: @"numberOfSimultaneousBookings"]
+                            forKey: @"numberOfSimultaneousBookings"];
+
+          // Get sAMAccountName if exists (LdapSource)
+          if ([userEntry objectForKey: @"samaccountname"])
+            [currentUser setObject: [userEntry objectForKey: @"samaccountname"]
+                            forKey: @"sAMAccountName"];
```

Comment:
Precisely I've used this name because this is AD specific. There is already quite a mess with naming: username, uid, c_uid and not in one place you really know what you have in that variable so I disagree and I'd prefer to keep this name